library/swift/src/ResponseHandler.swift

```diff
@@ -63,7 +63,7 @@ public final class ResponseHandler: NSObject {
   /// - parameter closure: Closure which will be called when an error occurs.
   @discardableResult
   public func onError(_ closure:
-    @escaping () -> Void)
+    @escaping (_ error: EnvoyError) -> Void)
```

Comment:
having this type here is causing swift compilation to segfault. I think I am doing things correctly in objc with the interface, implementation, and including the new EnvoyError.m file in the BUILD file. 

@rebello95 @goaway I would appreciate your eyes on this, I am sure I am missing something silly

Reply:
This looks correct to me - we previously saw some segfaults due to issues with the Swift compiler rather than actual bugs in our code, so this could be the same thing. Do you hit the same thing if you try passing an `NSError` instead of an `EnvoyError`?